### ISSUE TITLE
Maybe have async and blocking calls for `CrawURL`? 

### ISSUE THREAD

**Issue by rusenask** [mendableai/firecrawl-go#4]
The problem with:

```go
func (app *FirecrawlApp) CrawlURL(url string, params map[string]any, waitUntilDone bool, pollInterval int, idempotencyKey string) (any, error) {
```
is that it returns `any` which is something unexpected in Go. 

How about:

```go
func (app *FirecrawlApp) AsyncCrawlURL(url string, params map[string]any, waitUntilDone bool, pollInterval int, idempotencyKey string) (string, error) {
```
where it returns the job ID.

Then, for the sync option just return the whole thing. What do you think? I can make the pr with the cleanup

**Comment by rafaelsideguide** [comment]
@rusenask awesome suggestion! Going to implement that for v1 (probably on all SDKs too haha).

**Comment by KentHsu** [comment]
Hi @rusenask, totally agree that having separate methods can make things cleaner. Just curious about the naming convention, is it common to use an 'Async' prefix in function names in Go? Users can run `CrawlURL` in a goroutine and use a channel to get the result. There's no need to use an async prefix in function name to distinguish the sync/async behavior.

```go
func main() {
    resultChan := make(chan []*FirecrawlDocument)
    go func() {
        resultChan <- CrawlURL(...)  // set waitUntilDone = true
    }()
    
    // Do other work here if needed
    
    result := <-resultChan
}
```

**Comment by rafaelsideguide** [comment]
Closed by #3
